<!-- Launch file for L shape box fitting -->  
<launch>
  <arg name="namespace" default="l_shape_fitting"/>
  <arg name="packagename" default="autoware_bboxs_array"/>

  <arg name="input_topic" default="/segmentation/detected_objects" /><!--CHANGE THIS TO READ WHETHER FROM VSCAN OR POINTS_RAW -->
  <arg name="output_topic" default="/detection/lidar_objects/l_shaped" />

  <node pkg="ransac_lshape_fitting" type="ransac_lshape_fitting" name="ransac_lshape_fitting" output="screen" >
    <remap from="/segmentation/detected_objects" to="$(arg input_topic)" />
    <remap from="/detection/lidar_objects/l_shaped" to="$(arg output_topic)" />
  </node>

<!-- /l_shape_fitting/autoware_bboxs_array -->

  <node pkg="detected_objects_visualizer" type="visualize_detected_objects" name="shape_estimation_visualization"
        output="screen" ns="/$(arg namespace)/$(arg packagename)"/>
        
  <!-- Rviz -->
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find ransac_lshape_fitting)/rviz/ransac_l_shape_fitting.rviz" output="log" respawn="false" />

</launch>
